apps/omg_watcher/test/omg_watcher/api/core_test.exs

```diff
@@ -0,0 +1,104 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.API.CoreTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: true
+  use OMG.Fixtures
+
+  alias OMG.Block
+  alias OMG.State.Transaction
+  alias OMG.TestHelper
+  alias OMG.Utxo
+  alias OMG.Watcher.API.Core
+  require Utxo
+
+  @eth OMG.Eth.RootChain.eth_pseudo_address()
+
+  test "getting exit data returns error when there is no deposit" do
+    assert {:error, :no_deposit_for_given_blknum} == Core.compose_deposit_exit(nil, Utxo.position(1003, 0, 0))
+  end
+
+  @tag fixtures: [:alice]
+  test "creating deposit exit", %{alice: alice} do
+    position = Utxo.position(1000, 0, 0)
+    encode_utxo = position |> Utxo.Position.encode()
+
+    assert {:ok,
+            %{
+              utxo_pos: ^encode_utxo,
+              txbytes: _txbytes,
+              proof: _proof
+            }} =
+             Core.compose_deposit_exit(
+               %{
+                 amount: 10,
+                 currency: OMG.Eth.zero_address(),
+                 owner: alice.addr
+               },
+               position
+             )
+  end
+
+  @tag fixtures: [:alice]
+  test "compose output exit", %{alice: alice} do
+    tx_encode = fn number ->
+      TestHelper.create_signed([{1_000, number, 0, alice}], @eth, [{alice, 10}])
```

Comment:
I believe we have test helper [`create_encoded`](https://github.com/omisego/elixir-omg/blob/64d02e994ea06a9c8cda3650f5894050c7eacffa/apps/omg/test/support/test_helper.ex#L94), but this function is still helpful